generate.py

```diff
@@ -1,23 +1,36 @@
 import pandas as pd
+import requests
 from jinja2 import Environment, FileSystemLoader
 
+good_reads_api_key = input("Enter Goodreads Api Key: ")
 
 books_df = pd.read_excel("./input/Free+English+textbooks.xlsx")
 grouped_books_df = books_df.groupby(["English Package Name"])
 
-
 loader = FileSystemLoader(searchpath="./templates/")
 env = Environment(loader=loader)
 template = env.get_template("index.html")
 
+isbn_hyphen_maps = [{}, {}]
+isbn_review_map = {}
+
+for book in books_df.iterrows():
+    isbn_hyphen_maps[0][book[1]["Print ISBN"]] = book[1]["Print ISBN"].replace('-', '')
+    isbn_hyphen_maps[1][book[1]["Electronic ISBN"]] = book[1]["Electronic ISBN"].replace('-', '')
+for map in isbn_hyphen_maps:
+    response = requests.get("https://www.goodreads.com/book/review_counts.json?isbns=" + ','.join(map.values()) + 'key=' + good_reads_api_key)
+    for review in response.json()["books"]:
+        isbn_review_map[review["isbn13"]] = review
+
 grouped_books = {}
 
 for group, books in grouped_books_df:
     grouped_books[group] = []
     for book in books.iterrows():
+        book[1]["goodreads_review"] = isbn_review_map.get(isbn_hyphen_maps[0][book[1]["Print ISBN"]], isbn_review_map.get(isbn_hyphen_maps[1][book[1]["Electronic ISBN"]], None))
```

Comment:
Could you somehow add this as a column to the original dataframe before this loop?

Reply:
I cannot, I believe the `book` variable is of type `tuple` which cannot be dynamically added to.
![image](https://user-images.githubusercontent.com/7141342/85121962-a5bf8380-b1da-11ea-9a74-4189decc172c.png)

Reply:
Hm, after re-reading this a few times, I'm not sure I understand. Do you mean adding the columns we require directly to the `book` object, rather than storing them in the child `goodreads_review` object?